packages/app/src/settings/reducers/settings.reducer.test.ts

```diff
@@ -0,0 +1,386 @@
+  test("Event: SetLatestVersion set `latestVersion` value", () => {
+    expect(
+      settingsReducer(undefined, {
+        type: SettingsEvent.SetLatestVersion,
+        payload: "1.0.0",
+      })
+    ).toEqual({
+      ...initialState,
+      latestVersion: "1.0.0",
+    })
+  })
+})
+
+describe("Functionality: toggle tethering", () => {
+  test("Event: ToggleTethering/fulfilled set `tethering` value", () => {
+    expect(
+      settingsReducer(undefined, {
+        type: fulfilledAction(SettingsEvent.ToggleTethering),
+        payload: true,
+      })
+    ).toEqual({
+      ...initialState,
+      tethering: true,
+    })
+    expect(
+      settingsReducer(undefined, {
+        type: fulfilledAction(SettingsEvent.ToggleTethering),
+        payload: false,
+      })
+    ).toEqual({
+      ...initialState,
+      tethering: false,
+    })
+  })
+})
+
+describe("Functionality: toggle update available", () => {
+  test("Event: ToggleUpdateAvailable/pending set loading state", () => {
+    expect(
+      settingsReducer(undefined, {
+        type: pendingAction(SettingsEvent.ToggleUpdateAvailable),
+      })
+    ).toEqual({
+      ...initialState,
+      loading: true,
+      loaded: false,
+    })
+  })
+
+  test("Event: ToggleUpdateAvailable/fulfilled set updateAvailable value and changing loading state", () => {
+    expect(
+      settingsReducer(undefined, {
+        type: fulfilledAction(SettingsEvent.ToggleUpdateAvailable),
+        payload: true,
+      })
+    ).toEqual({
+      ...initialState,
+      updateAvailable: true,
+      loading: false,
+      loaded: true,
+    })
+    expect(
+      settingsReducer(undefined, {
+        type: fulfilledAction(SettingsEvent.ToggleUpdateAvailable),
+        payload: false,
+      })
+    ).toEqual({
+      ...initialState,
+      updateAvailable: false,
+      loading: false,
+      loaded: true,
+    })
+  })
+})
+
+describe("Functionality: toggle data collection", () => {
+  test("Event: ToggleCollectionData/fulfilled set `collectingData` value", () => {
+    expect(
+      settingsReducer(undefined, {
+        type: fulfilledAction(SettingsEvent.ToggleCollectionData),
+        payload: true,
+      })
+    ).toEqual({
+      ...initialState,
+      collectingData: true,
+    })
+    expect(
+      settingsReducer(undefined, {
+        type: fulfilledAction(SettingsEvent.ToggleCollectionData),
+        payload: false,
+      })
+    ).toEqual({
+      ...initialState,
+      collectingData: false,
+    })
+  })
+})
+
+describe("Functionality: check for update", () => {
+  test("Event: CheckUpdateAvailable/fulfilled set `updateAvailable` value", () => {
+    expect(
+      settingsReducer(
+        {
+          ...initialState,
+          updateAvailable: true,
+        },
+        {
+          type: fulfilledAction(SettingsEvent.CheckUpdateAvailable),
+          payload: true,
+        }
+      )
+    ).toEqual({
+      ...initialState,
+      updateAvailable: undefined,
```

Comment:
`undefined`? Is it ok?

Reply:
yep, the MC check if the value is `undefined` before. It's mean's that the update checking isn't initializing yet